File: benchmarks/ruby/tail-recursion.rb
def factorial(n)
  def fact_iter(n, acc)
    if n <= 1
      acc
    else
      fact_iter(n - 1, n * acc)
    end
  end
  fact_iter(n, 1)
end

def countdown(n)
  if n <= 0
    0
  else
    countdown(n - 1)
  end
end

def tail_rec_benchmark
  factorial(1000)
  countdown(10000)
end

tail_rec_benchmark

File: benchmarks/python/tail-recursion.py
def factorial(n):
    acc = 1
    while n > 1:
        acc *= n
        n -= 1
    return acc

def countdown(n):
    while n > 0:
        n -= 1
    return 0

def tail_rec_benchmark():
    factorial(1000)
    countdown(10000)

tail_rec_benchmark()